### PR TITLE
Feat(plugins): Support braces in range_expand filter

### LINKED ISSUE #2768: Add parenthesis syntax for interface ranges
### Enhancement summary

Some platforms have these kind of breakout layout

```
network_ports:
  - switches:
      - MY-AWESOME-SWITCHES
    switch_ports:
      # Special ports
      # Ethernet 26, 28, 30 and 32 have only 1 and 3
      # Ethernet 33 and 35 have from 1-8
      - Ethernet9-25/1-4
      - Ethernet26/1
      - Ethernet26/3
      - Ethernet27/1-4
      - Ethernet28/1
      - Ethernet28/3
      - Ethernet29/1-4
      - Ethernet30/1
      - Ethernet30/3
      - Ethernet31/1-4
      - Ethernet32/1
      - Ethernet32/3
      - Ethernet33/1-8
      - Ethernet34/1-4
      - Ethernet35/1-8
      - Ethernet36/1-4
 ```

Adding parenthesis would help reduce the snippet above to something like

```
network_ports:
  - switches:
      - MY-AWESOME-SWITCHES
    switch_ports:
      # Special ports
      # Ethernet 26, 28, 30 and 32 have only 1 and 3
      # Ethernet 33 and 35 have from 1-8
      Ethernet(9-25,27,29,31,34)/1-4
      Ethernet(26,28,30,32)/(1,3)
      Ethernet(33,35)/1-8
```

### Which component of AVD is impacted

plugins

### Use case example

cf description

### Describe the solution you would like

parenthesis for ranges - cf description

### Describe alternatives you have considered

current functionality is working fine

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Support braces in range_expand filter

## Component(s) name

`arista.avd.range_expand`

## Proposed changes

Add support for braces with ranges or comma-separated numbers:
```yaml
- "{{ 'eth{7,9,11-13}/1,21/1,26/1' | range_expand }}"
# -> ["eth7/1", "eth9/1", "eth11/1", "eth12/1", "eth13/1", "eth21/1", "eth26/1"]```
```

## How to test

Added complete unit tests for the rewritten python code and added to the existing unit tests for the Jinja filter.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
I will do some code improvements, so moving to draft.

File: ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    def __int__(self):
+        return int(self.value)
+
+
+class StringToken(DynamicToken):
+    """Token for something we don't expand like a prefix or a dot."""
+
+    @classmethod
+    def is_token(cls, value: str) -> bool:
+        # We accept "-"" if it is in a string to accept things like
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py]: comment looks incomplete?

File: ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    if isinstance(input_ranges, str):
+        input_ranges = [input_ranges]
+    elif not isinstance(input_ranges, list):
+        raise TypeError(f"range_expand only accepts a string or a list. Got {type(input_ranges)}.")
+
+    output = []
+    for input_range in input_ranges:
+        if not isinstance(input_range, str):
+            raise TypeError(f"range_expand only accepts input_range items as strings. Got {type(input_range)}.")
+        # raise Exception(_parser(_tokenizer(input_range)))
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py]: Is this required?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py]:
```suggestion
```

**Review by Vibhu-gslab — APPROVED** [review]
Only one comment looks incomplete (https://github.com/aristanetworks/avd/pull/3244/files#diff-fc897bdcc94b99674be5c963aa84a2c67fff033e684b3a7e767d92ec065e1e68R48)
otherwise looks good.
LGTM!

**Comment by Vibhu-gslab** [comment]
Also, ansible-lint CI check is failing.

File: ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py]:
```suggestion
        # We accept "-"" if it is in a string to accept things like "Port-Channel1-2"
```

**Review by Vibhu-gslab — APPROVED** [review]
LGTM!

**Review by carlbuchmann — APPROVED** [review]
LGTM, scale tested with very large input/outputs. although there is a small performance penalty IMO it's still very fast with normal types of input.

**Pull request merged**